dyson.js

```diff
@@ -102,11 +110,21 @@ function init(layers /* array of mathjs sparse matrices */,
 
     renderer.setClearColor( scene.fog.color );
     renderer.setPixelRatio( window.devicePixelRatio );
-    renderer.setSize( window.innerWidth, window.innerHeight );
+    renderer.setSize( innerWidth, innerHeight );
 
-    document.body.appendChild(renderer.domElement);
+    if (elementId) {
+        document.getElementById(elementId).appendChild(renderer.domElement);
+    } else {
+        document.body.appendChild(renderer.domElement);
+    }
 
-    window.addEventListener( 'resize', onWindowResize, false );
+    window.addEventListener('resize', function() {
+        camera.aspect = innerWidth / innerHeight;
+        camera.updateProjectionMatrix();
+        renderer.setSize( innerWidth, innerHeight );
+        controls.handleResize();
+        render();
+    }, false );
```

Comment:
Moved this to an inline function so it can have scope to `innerWidth` and `innerHeight`.